test/unit/methods/analyzing/addRank.test.ts

```diff
@@ -0,0 +1,195 @@
+import assert from "assert"
+import addRank from "../../../../src/methods/analyzing/addRank.js"
+
+describe("addRank", function () {
```

Comment:
A few tests to add. If the key doesn't exist. If the newKey already exists. Sorting values ascending, descending. And all ties options.